src/platform/plugins/shared/workflows_management/common/lib/build_execution_graph/build_execution_graph.ts

```diff
@@ -164,12 +175,13 @@ function visitForeachStep(graph: graphlib.Graph, previousStep: any, currentStep:
   return exitForeachNode;
 }
 
-export function convertToWorkflowGraph(workflow: WorkflowExecutionEngineModel): graphlib.Graph {
+export function convertToWorkflowGraph(workflowSchema: WorkflowSchema): graphlib.Graph {
   const graph = new graphlib.Graph({ directed: true });
-  let previousStep: BaseStep | null = null;
+  let previousNode: any | null = null;
```

Comment:
should we keep more specific type here?

Reply:
yes, but I don't have kind of "basenode" at the moment.
Let's keep it like this and in the future we will add it.
Anyway, tests verifie that the logic works right.